### PR TITLE
fix(48918): add description of missing third parameter `value` of `Reflect.set`.

### LINKED ISSUE #48918: Reflect: jsdoc about `Reflect.set`
# lib Update Request

## Configuration Check

My compilation *target* is `ES2015` and my *lib* is `the default`.

## Missing / Incorrect Definition

When I use `Reflect.set` to set a property on an object, I realize the unusual hint from TypeScript:

<details>
    <summary>
    Here is a screenshot from the TypeScript playground.
    </summary>
    image

image

image

image
</details>

When passing in the third parameter, the hint is lost.

Because of the experience of Issue https://github.com/microsoft/TypeScript/issues/48557, I realized that this may be a JSDoc problem again, so I looked at the source code and found that there is indeed one parameter definition missing.

Next I will try to fix it.

## Sample Code

```typescript
const obj = {};

Reflect.set(obj, 'key', 'value');
```

### PR DESCRIPTION
**Pull request by dev-itsheng**
Like PR https://github.com/microsoft/TypeScript/pull/48560, I Added description of missing third parameter `value`.

Fixes #48918

### CODE REVIEW

File: src/lib/es2015.reflect.d.ts [hunk 1]
```diff
@@ -87,6 +87,7 @@ declare namespace Reflect {
     /**
      * Sets the property of target, equivalent to `target[propertyKey] = value` when `receiver === target`.
      * @param target Object that contains the property on itself or in its prototype chain.
+     * @param value The value to set.
```
Comment (andrewbranch) [inline src/lib/es2015.reflect.d.ts:90]: It should come after `propertyKey`, not before

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Comment by andrewbranch** [comment]
It seems to me that this could have been absent because the description does not provide any more value than the param name. “The value to set” tells me the exact same amount of information as `value`. I’m not sure whether we should accept this.

File: src/lib/es2015.reflect.d.ts (hunk 1 above)
Comment (dev-itsheng) [reply src/lib/es2015.reflect.d.ts]: Thanks for pointing out such a detailed question, I didn't pay attention to the order of the parameters, it's fixed now.

**Comment by dev-itsheng** [comment]
> It seems to me that this could have been absent because the description does not provide any more value than the param name. “The value to set” tells me the exact same amount of information as `value`. I’m not sure whether we should accept this.

See [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/set).

When designing the description for this parameter, I referred to the description in the link above, and although it does look like "nothing is said", it at least completes the description of the parameter.

Before that, I noticed that in TypeScript's lib library, some methods have no parameter type description (ie `@params`), and some have, I think they should be unified: either each parameter has a description, Either each parameter is not stated. In the last PR, I tried to fill in the `littleEndian` related parameters.

---

Since you mentioned this, I feel it is necessary too. I also tried to explain the meaning of this parameter in more detail. So I found [MDN's description of the `value` property in `Object.defineProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#description):

> The value associated with the property. Can be any valid JavaScript value (number, object, function, etc). Defaults to undefined.

Remove the last sentence (because `value` is required in TypeScript), so can we use the first two sentences here?

**Comment by sandersn** [comment]
I agree with @andrewbranch that this doesn't add much at all -- and MDN's second sentence is also redundant given that `value` is typed as `any` already.

I'd prefer to be extremely conservative with the standard lib files.

**Comment by sandersn** [comment]
To help with PR housekeeping, I'm going to close this PR while it's still waiting on its bug to be accepted.

**Pull request closed without merging**